libcontainer/rootfs_linux.go

```diff
@@ -158,6 +180,19 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 				return err
 			}
 		}
+		if copyUp {
+			f, err := os.Open(tmpFilePath)
+			if err != nil {
+				return fmt.Errorf("failed to open %v for reading", tmpFilePath)
+			}
+			defer f.Close()
+			if err := archive.UntarUncompressed(f, dest, nil); err != nil {
+				return fmt.Errorf("failed to uncompress tar stream: %v", err)
+			}
+			if err := os.Remove(tmpFilePath); err != nil {
```

Comment:
It feels like some of this should be deferred (especially the removal of the archive).